File: leetcode-solutions/0242-valid-anagram/solution.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if(s.length() != t.length()){
            return false;
        }
        unordered_map<char,int> anag;
        for( char c: s){
            anag[c]++;
        }
        for(char c:t){
            if(anag[c] ==0){
                return false;
            }
            anag[c]--;
        }
        return true;
    }
};

File: leetcode-solutions/0242-valid-anagram/solution.py
from collections import Counter 
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        count = Counter(s)
        for char in t:
            if count[char]==0:
                return False
            count[char] -=1
        return True
